app/contracts/settings/working_days_and_hours_params_contract.rb

```
@@ -0,0 +1,96 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Settings
+  class WorkingDaysAndHoursParamsContract < ::ParamsContract
+    include RequiresAdminGuard
+
+    validate :working_days_are_present
+    validate :hours_per_day_are_present
+    validate :days_per_week_are_present
+    validate :days_per_month_are_present
+    validate :days_per_week_and_days_per_month_are_consistent
```

It does not validate enough IMHO:
- It should not be possible to set more than 24h per day, 7 days per week, or 31 days per month. It should raise an error.
- It should not be possible to enter negative values
- It should not be possible to enter zero values or text. Using text is "blocked" client side, but not checked server side. When doing so with a curl request, the value is interpreted as 0 due to some `to_i` or `to_f` here and there.
  - When hours per day value is 0, it leads to division by zero errors when converting time